packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkConfigurationExt.kt

```diff
@@ -55,6 +65,78 @@ internal fun ReadableMap.asCustomEndpoints(): CustomEndpoints {
     )
 }
 
+@Suppress("ComplexMethod")
+internal fun ReadableMap.asProxyConfig(): Pair<Proxy, ProxyAuthenticator?>? {
+    var address: String? = null
+    if (hasKey("address")) {
+        address = getString("address")
+    }
+    var port: Int? = null
+    if (hasKey("port")) {
+        port = getDouble("port").toInt()
+    }
+    val type =
```

Comment:
Perhaps ww to extract some of these to their own methods - e.g. a resolveProxyType method. wdyt?

Reply:
I think the logic is quite simple still and should not be modified very often, I think it's still ok to have it in one place to make it easier to understand.